internal/logging/logger.go

```diff
@@ -36,6 +37,10 @@ func NewLogger(writer io.Writer) *logger {
 }
 
 func UseServerLogger() {
+	// If the server is run from an interactive terminal, use the default ConsoleWriter
+	if os.Stdin != nil && term.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
We had this before #2432, I think it must have been lost in that change. If a server or connector is run with an interactive terminal then we want the ConsoleLogger.

This is different from short lived CLI commands like `infra grants list`. If short lived CLI commands are run without an interactive terminal we still want a ConsoleLogger, but we want color to be disabled (fixed in a later commit).